packages/shorebird_cli/lib/src/artifact_builder/artifact_builder.dart

```diff
@@ -93,22 +96,36 @@ class ArtifactBuilder {
         'build',
         'appbundle',
         '--release',
+        '-v',
         if (flavor != null) '--flavor=$flavor',
         if (target != null) '--target=$target',
         if (targetPlatformArgs != null) '--target-platform=$targetPlatformArgs',
         ...args,
       ];
 
-      final result = await process.run(
+      final spawnedProcess = await process.start(
         executable,
         arguments,
         runInShell: true,
         environment: base64PublicKey?.toPublicKeyEnv(),
       );
 
-      if (result.exitCode != ExitCode.success.code) {
+      final sub = spawnedProcess.stdout.map(utf8.decode).listen((event) {
+        progressUpdater?.onLog(event);
+      });
```

Comment:
This always subscribes even if there isn't a progressUpdater.  I wonder if it's better to only subscribe if there is one.  Then the listen call is just listen(progressUpdater.onLog). 🤷

Reply:
yeah, I had that wondering when I wrote this, I ended with this because I felt like the conditions that would needed to be added would make the code harder to read, with little benefit. I will put more thought on that either way since this will change with the suggestion Bryan gave below.

Reply:
I refactored a bunch of stuff in the PR and this went away in the changes, lmk wdyt